lib/error-handler.js

```diff
@@ -62,13 +63,31 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const result = func(error, reply.request, reply)
-  if (result !== undefined) {
-    if (result !== null && typeof result.then === 'function') {
-      wrapThenable(result, reply)
-    } else {
-      reply.send(result)
+  try {
+    const result = func(error, reply.request, reply)
+    if (result !== undefined) {
+      if (result !== null && typeof result.then === 'function') {
+        wrapThenable(result, reply)
+      } else {
+        reply.send(result)
+      }
+    }
+  } catch (err) {
+    reply[kReplyIsError] = true
+    try {
+      if (err.cause) {
+        throw new Error('Child error already has a cause set')
+      }
```

Comment:
I'm over the fence on this one; that can cause more confusion when dealing with unexpected errors.

In the scenario that I have a situation like this, where my `err.cause` already is set, I might prefer to have another wrapper on top that preserves the stack trace so I can fully track down the issue instead of throwing a new error that swallows the original one.

Maybe we can find a better strategy, but not sure throwing and swallowing might be the best way to go

Reply:
Can you clarify what you think is swallowing the error? It'd be the custom error handler implementation that could overwrite the cause - otherwise users would get the full stack trace by default.